plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/preferences/OpenShiftCorePreferences.java

```diff
@@ -61,7 +63,11 @@ private StringPreferenceValue createPreferenceValue(String connectionURL) {
 	}
 	
 	public String getOCBinaryLocation() {
-		return ocBinaryLocation.get();
+		String result = ocBinaryLocation.get();
+		if(StringUtils.isEmpty(result)) {
```

Comment:
that's the change I wouldnt do because it would change the scope of OpenShiftCorePreferences. It would not only get/store values in the preferences but now also look up in the path. I suggest to change OCBinary(Name) to look up in path if there's nothing in preferences.